tests/pipelines/test_pipelines_video_classification.py

```diff
@@ -0,0 +1,94 @@
+# limitations under the License.
+
+import unittest
+
+from huggingface_hub import hf_hub_download
+from transformers import MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING, VideoMAEFeatureExtractor
+from transformers.pipelines import VideoClassificationPipeline, pipeline
+from transformers.testing_utils import (
+    nested_simplify,
+    require_decord,
+    require_tf,
+    require_torch,
+    require_torch_or_tf,
+    require_vision,
+)
+
+from .test_pipelines_common import ANY, PipelineTestCaseMeta
+
+
+@require_torch_or_tf
+@require_vision
+@require_decord
+class VideoClassificationPipelineTests(unittest.TestCase, metaclass=PipelineTestCaseMeta):
+    model_mapping = MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING
+
+    def get_test_pipeline(self, model, tokenizer, feature_extractor):
+        example_video_filepath = hf_hub_download(
+            repo_id="nateraw/video-demo", filename="archery.mp4", repo_type="dataset"
+        )
+        video_classifier = VideoClassificationPipeline(model=model, feature_extractor=feature_extractor, top_k=2)
+        examples = [
+            example_video_filepath,
+            "https://huggingface.co/datasets/nateraw/video-demo/resolve/main/archery.mp4",
+        ]
+        return video_classifier, examples
+
+    def run_pipeline_test(self, video_classifier, examples):
+
+        for example in examples:
+            outputs = video_classifier(example)
+
+            self.assertEqual(
+                outputs,
+                [
+                    {"score": ANY(float), "label": ANY(str)},
+                    {"score": ANY(float), "label": ANY(str)},
+                ],
+            )
+
+    @require_torch
+    def test_small_model_pt(self):
+        small_model = "hf-internal-testing/tiny-random-VideoMAEForVideoClassification"
+        small_feature_extractor = VideoMAEFeatureExtractor(size=dict(shortest_edge=10), crop_size=dict(height=10, width=10))
+        video_classifier = pipeline(
+            "video-classification", model=small_model, feature_extractor=small_feature_extractor
+        )
+
+        video_file_path = hf_hub_download(repo_id="nateraw/video-demo", filename="archery.mp4", repo_type="dataset")
+        outputs = video_classifier(video_file_path, top_k=2)
+        self.assertEqual(
+            nested_simplify(outputs, decimals=4),
+            [{"score": 0.5209, "label": "LABEL_0"}, {"score": 0.4791, "label": "LABEL_1"}],
+        )
+
+        outputs = video_classifier(
+            [
+                video_file_path,
+                video_file_path,
+            ],
+            top_k=2,
+        )
+        self.assertEqual(
+            nested_simplify(outputs, decimals=4),
+            [
+                [{"score": 0.5209, "label": "LABEL_0"}, {"score": 0.4791, "label": "LABEL_1"}],
+                [{"score": 0.5209, "label": "LABEL_0"}, {"score": 0.4791, "label": "LABEL_1"}],
+            ],
+        )
+
+    @require_tf
+    def test_small_model_tf(self):
+        pass
```

Comment:
Do we have a large model to make sure the test actually works as intended and classifies this video as archery ?

Reply:
The default model [MCG-NJU/videomae-base-finetuned-kinetics](https://huggingface.co/MCG-NJU/videomae-base-finetuned-kinetics) works for this.

Reply:
Perfect let's add a slow test then.

Reply:
I believe the `run_pipeline_test` is doing this already? not sure. Looks like its grabbing the default model and doing this already